share/doc/CMakeLists.txt

```diff
@@ -2,9 +2,12 @@ set(DOXYGEN_SOURCES)
 foreach(DIR ${DOXYGEN_PACKAGES})
   set(ABS_DIR ${CMAKE_CURRENT_SOURCE_DIR}/../../${DIR})
   set(INPUT_DIRS "${INPUT_DIRS} ${ABS_DIR}")
-  file(GLOB_RECURSE ${DIR}_SOURCES ${ABS_DIR}/*.cc ${ABS_DIR}/*.h)
+  file(GLOB_RECURSE ${DIR}_SOURCES ${ABS_DIR}/*.cc ${ABS_DIR}/*.h
+         ${ABS_DIR}/*.dox ${ABS_DIR}/guide/*.md)
```

Comment:
@JoshuaSBrown: I added this, so the modules can include their own `dox` pages as well.